cwm_worker_operator/domains_config.py

```diff
@@ -409,9 +410,12 @@ def get_cwm_api_volume_config(self, metrics=None, force_update=False, hostname=N
         else:
             raise Exception("either hostname or worker_id param is required")
         start_time = common.now()
+        val = None
         if force_update or not worker_id:
-            # TODO: support cache for getting volume config based on hostname
-            val = None
+            cached_worker_id = self.keys.volume_config.get(hostname)
+            if cached_worker_id:
+                cached_worker_id = cached_worker_id.decode()
+                val = json.loads(self.keys.volume_config.get(cached_worker_id))
```

Comment:
the hostname and worker_id cache should not be dependant on each other

if worker_id is set - get the volume config from worker_id
if hostname is set - get the volume config from hostname
if force_update is true - don't use the cache

also, you should not handle decoding of internal Redis representation here. You should use a separate key of class `DomainsConfigKeyPrefixInt` which will handle this and return an int